pkgs/stdenv/generic/check-meta.nix

```
@@ -97,7 +97,7 @@ let
   hasUnfreeLicense = attrs: hasLicense attrs && isUnfree attrs.meta.license;
 
   hasNoMaintainers = attrs:
-    attrs ? meta.maintainers && (length attrs.meta.maintainers) == 0;
+    !(attrs ? meta.maintainers) || length attrs.meta.maintainers == 0;
```

Couldn't we simplify that to

```nix
```suggestion
    length attrs.meta.maintainers or [ ] == 0;
```

---

Or even `(attrs.meta.maintainers or [ ]) == [ ]`. [Nix uses a singleton for empty lists](https://github.com/nixos/nix/blob/45b0158d91499baf0782a828c7ea139445630c07/src/libexpr/eval.cc#L1329-L1331).